programs/server/Server.cpp

```diff
@@ -696,7 +696,7 @@ try
     OutdatedPartsLoadingThreadPool::initialize(
         server_settings.max_outdated_parts_loading_thread_pool_size,
         0, // We don't need any threads one all the parts will be loaded
-        server_settings.outdated_part_loading_thread_pool_queue_size);
+        server_settings.max_outdated_parts_loading_thread_pool_size);
```

Comment:
So, `max_threads` does not actually limit the number of threads in ThreadPool? Why does the queue size affect the number of threads?

Reply:
The problem is that `scheduleOrThrow` method throw an exception when there is no free slot in queue, which is 10k (makes no sense). We need such a behaviour to wait until there will be a free slot, which is guaranteed by using `scheduleOrThrowOnError`.